### PR TITLE
Introducing DNS rewrite (DNS DNATting) for inline mode

### PR DESCRIPTION
**Pull request by obilodeau**
owner: @dereksky 
assigned: @obilodeau

### CODE REVIEW

**Comment by obilodeau** [comment]
~~Not sure I like the fact to assign the DNAT to networks.conf's gateway parameter. This is not self-documenting.~~
We'll leave it that way for the moment. It is documented in PacketFence Administration Guide.

**Comment by obilodeau** [comment]
~~This feature would need a notice in UPGRADE.~~
Commit 1b99d16

**Comment by obilodeau** [comment]
~~Also, I think we talk in the documentation about the limitations of the inline mode (one VLAN only) and portal_redirect stuff. This would need to be removed / improved.~~
See Commits 6a40435 and dd39ae6

**Comment by fgaudreault** [comment]
~~Having a bug with this part I think :
Use of uninitialized value in pattern match (m//) at /usr/local/pf/lib/pf/util.pm line 119.
Use of uninitialized value in concatenation (.) or string at /usr/local/pf/lib/pf/services/named.pm line 104.
Use of uninitialized value in substitution iterator at /usr/local/pf/lib/pf/util.pm line 678, <$template_fh> line 13.~~
- Was trying to generate a template for a non-existing interface
- named was set to disabled
  See commit 5e2a86c and 623e4da

File: lib/pf/iptables.pm [hunk 1]
```diff
@@ -160,19 +160,18 @@ sub generate_inline_rules {
     my ($filter_rules_ref, $nat_prerouting_ref, $nat_postrouting_ref) = @_;
     my $logger = Log::Log4perl::get_logger('pf::iptables');
 
-    $logger->info("Allowing DNS through on inline interfaces to configured DNS servers");
-    foreach my $network (keys %ConfigNetworks) {
+    $logger->info("Adding DNS DNAT rules for unregistered and isolated inline clients.");
+    foreach my $network ( keys %ConfigNetworks ) {
         # skip non-inline interfaces
-        next if (!pf::config::is_network_type_inline($network));
+        next if ( !pf::config::is_network_type_inline($network) );
```
Comment (obilodeau) [inline lib/pf/iptables.pm:167]: lets try to not change whitespace only.. makes reviewing and merging more complex. (next if line and the foreach line)

**Comment by obilodeau** [comment]
Made minor changes. Now testing.

**Pull request merged**